grpc/src/client.rs

```diff
@@ -21,8 +21,11 @@ pub struct Client {
 
 impl Client {
     pub async fn connect(peer_discovery: impl PeerDiscovery) -> Result<Self, GrpcError> {
-        let url = Uri::from_str(peer_discovery.get_gapfiller_peer_addr().as_str())
-            .map_err(|e| GrpcError::UriCreate(e.to_string()))?;
+        Client::connect_to_url(peer_discovery.get_gapfiller_peer_addr().as_str()).await
```

Comment:
```suggestion
Client::connect_to_url(&peer_discovery.get_gapfiller_peer_addr()).await
```

just a preference of style, feel free to ignore